pkg/operator/starter.go

```diff
@@ -119,105 +154,138 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
-		"node.yaml",
-		kubeClient,
-		kubeInformersForNamespaces.InformersFor(defaultNamespace),
-		[]factory.Informer{configMapInformer.Informer()},
-		csidrivernodeservicecontroller.WithObservedProxyDaemonSetHook(),
-		csidrivernodeservicecontroller.WithCABundleDaemonSetHook(
-			defaultNamespace,
-			trustedCAConfigMap,
-			configMapInformer,
+			controlPlaneConfigMapInformer,
 		),
 	).WithServiceMonitorController(
 		"AWSEBSDriverServiceMonitorController",
-		dynamicClient,
-		assets.ReadFile,
+		controlPlaneDynamicClient,
+		assetWithNamespaceFunc(controlPlaneNamespace),
 		"servicemonitor.yaml",
 	).WithStorageClassController(
 		"AWSEBSDriverStorageClassController",
 		assets.ReadFile,
 		"storageclass_gp3.yaml",
-		kubeClient,
-		kubeInformersForNamespaces.InformersFor(""),
+		controlPlaneKubeClient,
+		controlPlaneKubeInformersForNamespaces.InformersFor(""),
 	)
 	if err != nil {
 		return err
 	}
 
-	caSyncController, err := newCustomCABundleSyncer(
-		operatorClient,
-		kubeInformersForNamespaces,
-		kubeClient,
-		controllerConfig.EventRecorder,
+	// Start controllers that manage resources in GUEST clusters.
+	guestCSIControllerSet := csicontrollerset.NewCSIControllerSet(
+		guestOperatorClient,
+		eventRecorder,
+	).WithStaticResourcesController(
+		"AWSEBSDriverGuestStaticResourcesController",
+		guestKubeClient,
+		guestDynamicClient,
+		guestKubeInformersForNamespaces,
+		assets.ReadFile,
+		[]string{
+			"storageclass_gp2.yaml",
+			"storageclass_gp3.yaml",
+			"volumesnapshotclass.yaml",
+			"csidriver.yaml",
+			"node_sa.yaml",
+			"rbac/node_privileged_binding.yaml",
+		},
+	).WithCSIDriverNodeService(
+		"AWSEBSDriverNodeServiceController",
+		assets.ReadFile,
+		"node.yaml",
+		guestKubeClient,
+		guestKubeInformersForNamespaces.InformersFor(guestNamespace),
+		[]factory.Informer{guestConfigMapInformer.Informer()},
+		csidrivernodeservicecontroller.WithObservedProxyDaemonSetHook(),
+		csidrivernodeservicecontroller.WithCABundleDaemonSetHook(
+			guestNamespace,
+			trustedCAConfigMap,
+			guestConfigMapInformer,
+		),
+	)
+
+	caSyncController, err := newCustomAWSBundleSyncer(
+		guestOperatorClient,
+		controlPlaneKubeInformersForNamespaces,
+		controlPlaneKubeClient,
+		controlPlaneNamespace,
+		eventRecorder,
 	)
 	if err != nil {
 		return fmt.Errorf("could not create the custom CA bundle syncer: %w", err)
 	}
 
-	klog.Info("Starting the informers")
-	go kubeInformersForNamespaces.Start(ctx.Done())
-	go dynamicInformers.Start(ctx.Done())
-	go configInformers.Start(ctx.Done())
+	klog.Info("Starting the control plane informers")
+	go controlPlaneKubeInformersForNamespaces.Start(ctx.Done())
+
+	klog.Info("Starting control plane controllerset")
+	go controlPlaneCSIControllerSet.Run(ctx, 1)
+
+	// Only start caSyncController in standalone clusters because in Hypershift
+	// the ConfigMap is already available in the correct namespace.
+	if !isHypershift {
+		klog.Info("Starting custom CA bundle sync controller")
```

Comment:
Move whole caSyncController creation above here? It's weird the controller is created and then not used in HyperShift.

Reply:
Fixed.